File: Lecture 2/Readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
int main(void){

    string user_text = get_string("Text:");
    int letters = 0;
    int words = 1;
    int sentences =0;
    for (int u =0; u<strlen(user_text); u++){
          if(isalpha(user_text[u])){
            letters++;
          } else if(user_text[u] == ' '){
             words++;
          } else if (user_text[u] == '.' || user_text[u] == '!' || user_text[u] == '?'){
            sentences+=1;
          }

    }
    float L1 = (float) letters / (float) words * 100;
    float S1 = (float) sentences / (float) words * 100;
    int index = round(0.0588 * L1 - 0.296 * S1 - 15.8);

    if(index < 1){
      printf("Before Grade 1\n");
    }

    else if (index > 16){;
      printf("Grade 16+\n");
    } else {
      printf("Grade %d\n", index);
    } 

}

File: Lecture 6/sentimental/readability/readability.py
import math
from cs50 import get_string


def readability(letters):
    text = get_string("Enter:")
    for i in text:

        if i.isalpha():
            letters += 1


    word_num = len(text.split())

    num_sentences = text.count(".") + text.count("!") + text.count("?")

    L1 = (float(letters) / float(word_num)) * 100
    S1 = (float(num_sentences) / float(word_num)) * 100

    index = round(0.0588 * L1 - 0.296 * S1 - 15.8)
    print(index)

    if (index < 1):
        print("Before Grade 1")

    if (index > 16):
        print("Grade 16+")

    else:
        print(f"Grade {index}")
readability( letters=0)


